src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -1030,7 +1031,251 @@ private void ReadCompressedTextChunk(PngMetadata metadata, ReadOnlySpan<byte> da
 
             if (this.TryUncompressTextData(compressedData, PngConstants.Encoding, out string uncompressed))
             {
-                metadata.TextData.Add(new PngTextData(name, uncompressed, string.Empty, string.Empty));
+                if (name.Equals("Raw profile type exif", StringComparison.OrdinalIgnoreCase) &&
+                    this.TryReadLegacyExifTextChunk(baseMetadata, uncompressed))
+                {
+                    // Successfully parsed exif data stored as text in this chunk
```

Comment:
Reverse the condition here so you don't have the odd empty scope.

Reply:
I ended up moving this into its own function, `TryReadTextChunkMetadata`, since I was missing support for loading legacy exif data from uncompressed text chunks, and I also wanted to have an easy spot to add support for other metadata chunks in the future. I suppose I could also just add IPTC support in this PR too if you want, but I figure it's getting somewhat large and it might be better to just make another PR with that feature after this one gets merged.

Reply:
Yeah, let's keep IPTC  separate for now.